wifi: esp32s3: Fix IRQ allocation

raffarost opened the pull request:
HAL code depends on ESP_WIFI_TASK_CORE_ID symbol to properly build the reserved IRQs table.
If define is not present, IRQ 0 (wifi) is not reserved, causing interrupt allocation conflicts.

edit: Kconfig symbol moved to HAL.

uLipe reviewed (commented):
Just a question, otherwise looks good

Commit:
wifi: esp32s3: Fix IRQ allocation

Update HAL to fix wifi IRQ allocation for ESP32-S3.
HAL code depends on ESP_WIFI_TASK_CORE_ID symbol to properly build
the reserved IRQs table. If define is not present, IRQ 0 (wifi) is
not reserved, causing interrupt allocation problems.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>
